db/dao/dao.go

```diff
@@ -123,6 +139,15 @@ func (dao *DAO[T]) Delete(ctx context.Context, id string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+	if dao.opts.CachePropagate && rows > 0 {
+		key := dao.opts.CacheKey.Build(id)
+		if e := mcache.Invalidate(ctx, key); e != nil {
+			dao.log.Warnf("failed to invalidate mcache: key=%s, %v", key, err)
+		}
+		dao.publishEvent(eventbus.EventInvalidation, map[string]interface{}{
```

Comment:
use go routine to publish if not care returning

Reply:
SGTM.